model/plotWidget.cc

```diff
@@ -488,21 +489,22 @@ namespace minsky
             }
           
           // higher rank y objects treated as multiple y vectors to plot
-          for (size_t j=0 /*d[0]*/; j<std::min(size_t(10)*d[0], yv.numElements()); j+=d[0])
-            {
-              setPen(extraPen, x, yv.begin()+j, d[0]);
-              if (pen>=numLines)
-                assignSide(extraPen,Side::right);
-             string label;
-              size_t stride=d[0];
-              for (size_t i=1; i<yv.xVector.size(); ++i)
-                {
-                  label+=str(yv.xVector[i][(j/stride)%d[i]])+" ";
-                  stride*=d[i];
-                }
-              labelPen(extraPen,label);
-              extraPen++;
-            }
+          // For feature 47
+            for (size_t j=0 /*d[0]*/; j<std::min(size_t(10)*d[0], yv.dataSize()); j+=d[0])
```

Comment:
In this case, we need to iterate j over the first 10 elements of the index vector.

Reply:
At present, I'm using the index vector just to propagate the size of the sparse data everywhere. I still need to modify the parts of the code which require the actual hypercube indices of the sparse data (e.g., CSV export and the code in plotWidget.cc you point out here).